production/docker/config/nginx-loki-gateway.conf

```diff
@@ -0,0 +1,64 @@
+error_log  /dev/stderr;
+pid        /tmp/nginx.pid;
+worker_rlimit_nofile 8192;
+
+events {
+    worker_connections  4096;  ## Default: 1024
+}
+
+http {
+
+  default_type application/octet-stream;
+  log_format   main '$remote_addr - $remote_user [$time_local]  $status '
+    '"$request" $body_bytes_sent "$http_referer" '
+    '"$http_user_agent" "$http_x_forwarded_for"';
+  access_log   /dev/stderr  main;
+  sendfile     on;
+  tcp_nopush   on;
+
+  upstream distributor {
+    server loki-1:3100;
+    server loki-2:3100;
+    server loki-3:3100;
+  }
+
+  upstream querier {
+    server loki-1:3100;
+    server loki-2:3100;
+    server loki-3:3100;
+  }
+
+  upstream query-frontend {
+    server loki-frontend:3100;
+  }
+
+  server {
+    listen 80;
+    proxy_set_header     X-Scope-OrgID 1;
+
+    location = /loki/api/v1/push {
+        proxy_pass       http://distributor$request_uri;
+    }
```

Comment:
I suggest to add also `/ring` for the distributor. It is a debugging endpoint for memberlist only available on the distributor.